packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -135,9 +135,12 @@ InternalModel.prototype = {
   },
 
   finishedReloading: function() {
-    this.isReloading = false;
-    if (this.record) {
-      set(this.record, 'isReloading', false);
+    var record = this;
+    if (record._loadingPromiseChain._state) {
```

Comment:
@igorT -- I thought that it was inherent in promises.

`_loadingPromiseChain` is a promise that keeps getting chained with new promises when a reload is called. A promise class sets the `_state` value when it completes either with an error or not. Since I'm always updating the `_loadingPromiseChain` to be the last promise on the chain, if that hasn't successfully finished (because someone else called `reload()`, then it doesn't change the flags back to false.

Did you want me to comment that in the code?